source/common/http/http2/codec_impl.cc

```diff
@@ -416,18 +397,17 @@ Status ConnectionImpl::StreamImpl::onDataSourceSend(const uint8_t* framehd, size
   parent_.protocol_constraints_.incrementOutboundDataFrameCount();
 
   Buffer::OwnedImpl output;
-  auto status = parent_.addOutboundFrameFragment(output, framehd, FRAME_HEADER_SIZE);
-  if (!status.ok()) {
+  parent_.addOutboundFrameFragment(output, framehd, FRAME_HEADER_SIZE);
+  if (!parent_.protocol_constraints_.checkOutboundFrameLimits().ok()) {
     ENVOY_CONN_LOG(debug, "error sending data frame: Too many frames in the outbound queue",
                    parent_.connection_);
     setDetails(Http2ResponseCodeDetails::get().outbound_frame_flood);
```

Comment:
Should there be an error return for this case, or should this be void and the call site always return 0?

Reply:
The error will be handled in the `sendPendingFrames()`. I've made this function void.